docs/tutorials/prefect.md

```diff
@@ -0,0 +1,90 @@
+# Integrating nnbench with Prefect
+
+If you have more complex workflows it is sensible to use a workflow orchestration tools. 
+Benchmarking with nnbench can be integrated with orchestrators. We will present an example integration with Prefect.
+We will explain the orchestration concepts in a high level and link to the corresponding parts of the  Prefect [docs](https://docs.prefect.io/).
+The full example code is in the nnbench [repository](https://github.com/aai-institute/nnbench/tree/main/examples/prefec).
+
+In this example we want to orchestrate the training and benchmarking of a linear regression model.
+
+## Project Structure
+### Defining the training tasks and workflows
+We recommend to separate the training and benchmarking logic. 
+
+```python
+--8<-- "examples/prefect/training.py"
+```
+
+The `training.py` file contains functions to generate synthetic data for our regression model, facilitate a train-test-split, and finally train the regression model.
+We have applied Prefects [`@task` decorator.](https://docs.prefect.io/latest/concepts/tasks/)  
+It marks the contained logic as a discrete unit of work for Prefect. 
+Two other functions prepare the regression data and train the estimator. 
+They are labeled with the [`@flow` decorator.](https://docs.prefect.io/latest/concepts/flows) which labels the function as a workflow logic container that can depend on other flows or tasks.
+The `prepare_regressor_and_test_data` function returns the model and test data so that we can use it in our benchmarks.
+
+### Defining Benchmarks 
+The benchmarks are in the `benchmark.py` file.
+We have two functions to calculate the mean absolute error and the mean squared error.
+These benchmarks are tagged to indicate they are metrics.
+The other two benchmarks calculate model metadata, namely the size and inference time of the model.
+
+```python
+--8<-- "examples/prefect/benchmark.py"
+```
+
+We did not apply any Prefect decorators here as they are incompatible with nnbench's decorators.
```

Comment:
True, but they are also a finer-grained unit of work, as we currently operate under the "one benchmark family per task" paradigm. I would mention that here instead.